src/post-primary-brand/index.js

```diff
@@ -0,0 +1,117 @@
+/* global newspackPostPrimaryBrandVars */
+
+import { __, _n, _x } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Button, Flex, FlexItem, SelectControl } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+import './index.scss';
+
+/**
+ * Module Constants
+ */
+const DEFAULT_QUERY = {
+	per_page: -1,
+	orderby: 'name',
+	order: 'asc',
+	_fields: 'id,name,parent',
+	context: 'view',
+};
+
+const EMPTY_ARRAY = [];
+
+const ADMIN_URL = newspackPostPrimaryBrandVars.adminURL;
+
+const TAXONOMY_SLUG = newspackPostPrimaryBrandVars.taxonomySlug;
+
+const META_KEY = newspackPostPrimaryBrandVars.metaKey;
+
+/**
+ * Adds a primary brand selector to the post editor.
+ */
+const NewspackPostPrimaryBrand = ( { slug } ) => {
+	const { editPost } = useDispatch( 'core/editor' );
+
+	const [ currentPrimaryBrand, setCurrentPrimaryBrand ] = useState( null );
+
+	const { terms, availableTerms, taxonomy } = useSelect(
+		select => {
+			const { getCurrentPost, getEditedPostAttribute } = select( 'core/editor' );
+			const { getTaxonomy, getEntityRecords } = select( coreStore );
+			const _taxonomy = getTaxonomy( slug );
+			const post = getCurrentPost();
+
+			setCurrentPrimaryBrand( post.meta[ META_KEY ] );
+
+			return {
+				terms: _taxonomy ? getEditedPostAttribute( _taxonomy.rest_base ) : EMPTY_ARRAY,
+				availableTerms: getEntityRecords( 'taxonomy', slug, DEFAULT_QUERY ) || EMPTY_ARRAY,
+				taxonomy: _taxonomy,
+			};
+		},
+		[ slug ]
+	);
+
+	const getTermSelectOptionFromId = id => {
+		const term = availableTerms.find( term => term.id === id );
+		return term ? { value: term.id, label: term.name } : null;
+	};
+
+	const onChangePrimaryBrand = termId => {
+		editPost( { meta: { [ META_KEY ]: termId } } );
+		setCurrentPrimaryBrand( termId );
+	};
+
+	return (
+		<Flex direction="column" gap="4">
+			<div
+				className="editor-post-taxonomies__hierarchical-terms-list"
```

Comment:
Is this class doing anything to the selector? I think we should have our own `newspack-multibranded-site__primary-brand__selector` class. It could use an 8px of top margin to not look so attached to the checkbox list:

<img width="283" alt="image" src="https://user-images.githubusercontent.com/820752/232879598-a9d5df65-113c-487d-a795-b2e6cff73711.png">

Reply:
done